File: Algorithm/Leetcode/0000_0100/0011_0020/0015.cpp
// 1281. Subtract the Product and Sum of Digits of an Integer

class Solution {
public:
    int subtractProductAndSum(int n) {
        int sum = 0;
        int multi = 1;
        while( n )
        {
            int pos = n % 10;
            sum += pos;
            multi *= pos;
            n /= 10;
        }
        return multi- sum;

    }
};

File: Algorithm/Leetcode/0000_0100/0011_0020/0015.py
# 1281. Subtract the Product and Sum of Digits of an Integer

"""https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/discuss/446372/JavaC%2B%2BPython-Straight-Forward-Solution"""

class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        A = list(map(int, str(n)))
        return reduce(operator.mul, A) - sum(A)